extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -24,123 +22,129 @@ const { naturalizeDataset, denaturalizeDataset } = DicomMetaDictionary;
+      implementation.clientCanReject = clientManager.clientCanReject();
     },
     query: {
       studies: {
         mapParams: mapParams.bind(),
         search: async function (origParams) {
-          qidoDicomWebClient.headers = getAuthrorizationHeader();
-          const { studyInstanceUid, seriesInstanceUid, ...mappedParams } =
-            mapParams(origParams, {
-              supportsFuzzyMatching: dicomWebConfig.supportsFuzzyMatching,
-              supportsWildcard: dicomWebConfig.supportsWildcard,
-            }) || {};
-
-          const results = await qidoSearch(qidoDicomWebClient, undefined, undefined, mappedParams);
-
+          clientManager.setQidoHeaders();
+          let results = [];
+          const studyInstanceUIDs = {};
+          // concatenate series metadata from all servers
+          const clients = clientManager.getClients();
+          for (let i = 0; i < clients.length; i++) {
+            const { studyInstanceUid, seriesInstanceUid, ...mappedParams } =
+              mapParams(origParams, {
+                supportsFuzzyMatching: clients[i].supportsFuzzyMatching,
+                supportsWildcard: clients[i].supportsWildcard,
+              }) || {};
+            let clientResults;
+            try {
+              clientResults = await qidoSearch(
+                clients[i].qidoDicomWebClient,
+                undefined,
+                undefined,
+                mappedParams
+              );
+            } catch {
+              clientResults = [];
+            }
+            for (let j = 0; j < clientResults.length; j++) {
+              const studyInstanceUID = clientResults[j][STUDY_INSTANCE_UID].Value[0];
+              if (!(studyInstanceUID in studyInstanceUIDs)) {
+                studyInstanceUIDs[studyInstanceUID] = clientResults[j];
+                results.push(clientResults[j]);
+              } else {
+                // updates number of series in study
+                if (
+                  studyInstanceUIDs[studyInstanceUID][NUMBER_STUDY_SERIES] &&
+                  clientResults[j][NUMBER_STUDY_SERIES]
+                ) {
+                  studyInstanceUIDs[studyInstanceUID][NUMBER_STUDY_SERIES].Value[0] =
+                    studyInstanceUIDs[studyInstanceUID][NUMBER_STUDY_SERIES].Value[0] +
+                    clientResults[j][NUMBER_STUDY_SERIES].Value[0];
+                }
+
+                // updates number of instances in study
+                if (
+                  studyInstanceUIDs[studyInstanceUID][NUMBER_STUDY_INSTANCES] &&
+                  clientResults[j][NUMBER_STUDY_INSTANCES]
+                ) {
+                  studyInstanceUIDs[studyInstanceUID][NUMBER_STUDY_INSTANCES].Value[0] =
+                    studyInstanceUIDs[studyInstanceUID][NUMBER_STUDY_INSTANCES].Value[0] +
+                    clientResults[j][NUMBER_STUDY_INSTANCES].Value[0];
+                }
+              }
+            }
```

Comment:
please refactor to a utility or something, this is noise in this file

Reply:
The code is wrong. It tries to fix the number of series and instances in the study level, but it depends on seriesinstanceuid